scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/core/CheckEligibleTaskDescriptorScriptTest.java

```diff
@@ -0,0 +1,245 @@
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.core;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.ow2.proactive.scheduler.common.SchedulerConstants;
+import org.ow2.proactive.scheduler.common.task.ForkEnvironment;
+import org.ow2.proactive.scheduler.common.task.flow.FlowScript;
+import org.ow2.proactive.scheduler.descriptor.EligibleTaskDescriptor;
+import org.ow2.proactive.scheduler.task.containers.ScriptExecutableContainer;
+import org.ow2.proactive.scheduler.task.internal.InternalScriptTask;
+import org.ow2.proactive.scheduler.task.internal.InternalTask;
+import org.ow2.proactive.scripting.InvalidScriptException;
+import org.ow2.proactive.scripting.Script;
+import org.ow2.proactive.scripting.SimpleScript;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 23/06/2017
+ */
+public class CheckEligibleTaskDescriptorScriptTest {
+
+    private EligibleTaskDescriptor etd;
+
+    private InternalTask it;
+
+    private InternalScriptTask ist;
+
+    private ScriptExecutableContainer sec;
+
+    private ForkEnvironment fe;
+
+    @Before
+    public void init() {
+        etd = mock(EligibleTaskDescriptor.class);
+        it = mock(InternalTask.class);
+        ist = mock(InternalScriptTask.class);
+        sec = mock(ScriptExecutableContainer.class);
+        fe = mock(ForkEnvironment.class);
+
+        Mockito.when(etd.getInternal()).thenReturn(it);
+        Mockito.when(ist.getExecutableContainer()).thenReturn(sec);
+        Mockito.when(it.getForkEnvironment()).thenReturn(fe);
+        Mockito.when(ist.getForkEnvironment()).thenReturn(fe);
+    }
+
+    @Test
+    public void testAllScriptsContainAPIBinding() throws InvalidScriptException {
+        Script s = scriptWithApiBindingClient();
+        Script s1 = scriptWithApiBindingUser();
+        Script s2 = scriptWithApiBindingGlobal();
+        FlowScript fs = flowScriptWithApiBindingClient();
+        Mockito.when(it.getPreScript()).thenReturn(s1);
+        Mockito.when(it.getPostScript()).thenReturn(s2);
+        Mockito.when(sec.getScript()).thenReturn(s);
+        Mockito.when(fe.getEnvScript()).thenReturn(s);
+        Mockito.when(it.getFlowScript()).thenReturn(fs);
+        Boolean result = new CheckEligibleTaskDescriptorScript().isTaskContainsAPIBinding(etd);
+        assertEquals(true, result);
```

Comment:
use assertTrue(result) or assertThat(result, isTrue) instead of assertEquals which is not relevant in this case.